server/src/com/cloud/vm/UserVmManagerImpl.java

```diff
@@ -910,35 +910,40 @@ private UserVm rebootVirtualMachine(long userId, long vmId) throws InsufficientC
                     startVirtualMachine(vmId, null, null, null);
                 }
             } else {
-                collectVmDiskStatistics(vm);
-                collectVmNetworkStatistics(vm);
```

Comment:
Should we leave these two lines to collect statistics before?

Reply:
It is buried about 10 layers deep, but the stopVirtualMachine() call does eventually get to the UserVmManagerImpl.prepareStop() method that does both of these already. It only existed here because reboots take a very different path through the system.